apps/public/public_test.go

```diff
@@ -859,36 +859,111 @@ func TestWithDomainHint(t *testing.T) {
 func TestWithAuthenticationScheme(t *testing.T) {
 	clientInfo := base64.RawStdEncoding.EncodeToString([]byte(`{"uid":"uid","utid":"utid"}`))
 	lmo, tenant := "login.microsoftonline.com", "tenant"
-	authori := fmt.Sprintf(authorityFmt, lmo, tenant)
 	accessToken, idToken, refreshToken := "at", mock.GetIDToken(tenant, lmo), "rt"
 	authScheme := mock.NewTestAuthnScheme()
+	var ar AuthResult
+	var client Client
+	var err error
 
-	mockClient := mock.Client{}
-	mockClient.AppendResponse(mock.WithBody(mock.GetTenantDiscoveryBody(lmo, tenant)))
-	mockClient.AppendResponse(mock.WithBody(mock.GetAccessTokenBody(accessToken, idToken, refreshToken, clientInfo, 3600)))
-	client, err := New("client-id", WithAuthority(authori), WithHTTPClient(&mockClient))
-	if err != nil {
-		t.Fatal(err)
+	for _, testCase := range []struct {
+		name      string
+		responses [][]byte
+	}{
+		{
+			name: "interactive",
+			responses: [][]byte{
+				mock.GetTenantDiscoveryBody(lmo, tenant),
+				mock.GetAccessTokenBody(accessToken, idToken, refreshToken, clientInfo, 3600),
+			},
+		},
+		{
+			name: "password",
+			responses: [][]byte{
+				mock.GetTenantDiscoveryBody(lmo, tenant),
+				[]byte(`{"account_type":"Managed","cloud_audience_urn":"urn","cloud_instance_name":"...","domain_name":"..."}`),
```

Comment:
Are you able to manually test this with a federated account?  (with a federated account, ROPC is really an OBO that exchanges a SAML token for a JWT token). You might be able to use your corp account for this, if not, I will follow up with a test account.